local/index.go

```diff
@@ -0,0 +1,249 @@
+package local
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/layout"
+	"github.com/google/go-containerregistry/pkg/v1/match"
+	"github.com/google/go-containerregistry/pkg/v1/mutate"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+)
+
+type ImageIndex struct {
+	docker   DockerClient
+	repoName string
+	index    v1.ImageIndex
+}
+
+var manifestDir string = "out/manifests"
+
+func (i *ImageIndex) Add(repoName string) error {
+	ref, err := name.ParseReference(repoName)
+	if err != nil {
+		panic(err)
+	}
+
+	desc, err := remote.Get(ref, remote.WithAuthFromKeychain(authn.DefaultKeychain))
+	if err != nil {
+		panic(err)
+	}
+
+	img, err := desc.Image()
+	if err != nil {
+		panic(err)
+	}
+
+	cfg, err := img.ConfigFile()
+
+	if err != nil {
+		return errors.Wrapf(err, "getting config file for image %q", repoName)
+	}
+	if cfg == nil {
+		return fmt.Errorf("missing config for image %q", repoName)
+	}
+	if cfg.OS == "" {
```

Comment:
I have removed the code validating `OS` and `Arch` and moved the validation to pack-cli so that the client won't allow pushing an index with incomplete platform information.